### ISSUE TITLE
Config files from nested folders

### ISSUE THREAD

**Issue by Aarkon** [adam-gaia/new-stow#1]
Greetings!

I have a dotfiles-repo roughly looking like this:

```
├── machineA
│   ├── stowfile
│   └── sway
│       └── config
├── helix
│   ├── config.toml
│   └── languages.toml
├── machineB
│   ├── stowfile
│   └── sway
│       └── config
├── machineC
│   └── stowfile
└── README.md
```

What doesn't work is stowing the `machine{A,B}/sway/config` files: 

```
---

stow:
  - src: ../helix/config.toml
    links:
      - "${HOME}/.config/helix/config.toml"

  - src: ../helix/languages.toml
    links:
      - "${HOME}/.config/helix/languages.toml"

  - sway:
    - src: config
      links:
        - "${HOME}/.config/sway/config"
  
  - src: ../.zshrc
    links:
      - "${HOME}/.zshrc" 
```
This gives me: `Error: Source file "/home/aarkon/src/dotfiles/machineA/../helix/../helix/sway/config.conf" does not exist" does not exist`. Without having looked at your code, I suggest the reason is that the path to the `sway/config` file gets wrongly extended. If you could fix that, I'd be an even more happy user of your nice tool :smiley:

**Comment by adam-gaia** [comment]
Hello,

I've written some tests to mimic your setup and have found the bug (a couple bugs actually!). It's nice to have someone else with a different setup than I expected to find issues. I'll have a fix + new release later today or tomorrow.

Sorry for the delay and thanks for using nstow!

**Comment by adam-gaia** [comment]
I've merged a fix and published a new version, `0.1.6`. 

Feel free to reach out if you need anything else :)
